<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MS2 fragment ions • PSMatch</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MS2 fragment ions">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">PSMatch</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/AdjacencyMatrix.html">Understanding protein groups with adjacency matrices</a>
    </li>
    <li>
      <a href="../articles/Fragments.html">MS2 fragment ions</a>
    </li>
    <li>
      <a href="../articles/PSM.html">Working with PSM data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RforMassSpectrometry/PSM/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MS2 fragment ions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RforMassSpectrometry/PSM/blob/main/vignettes/Fragments.Rmd" class="external-link"><code>vignettes/Fragments.Rmd</code></a></small>
      <div class="hidden name"><code>Fragments.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><p><strong>Package</strong>: <em><a href="https://bioconductor.org/packages/3.22/PSMatch" class="external-link">PSMatch</a></em><br><strong>Authors</strong>: Laurent Gatto [aut, cre] (ORCID: <a href="https://orcid.org/0000-0002-1520-2268" class="external-link uri">https://orcid.org/0000-0002-1520-2268</a>), Johannes Rainer
[aut] (ORCID: <a href="https://orcid.org/0000-0002-6977-7147" class="external-link uri">https://orcid.org/0000-0002-6977-7147</a>), Sebastian Gibb
[aut] (ORCID: <a href="https://orcid.org/0000-0001-7406-4443" class="external-link uri">https://orcid.org/0000-0001-7406-4443</a>), Samuel Wieczorek
[ctb], Thomas Burger [ctb], Guillaume Deflandre [ctb] (ORCID: <a href="https://orcid.org/0009-0008-1257-2416" class="external-link uri">https://orcid.org/0009-0008-1257-2416</a>)<br><strong>Last modified:</strong> 2025-08-02 07:09:52.050822<br><strong>Compiled</strong>: Sat Aug 2 07:24:18 2025</p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette is one among several illustrating how to use the
<code>PSMatch</code> package, focusing on the calculation and
visualisation of MS2 fragment ions. For a general overview of the
package, see the <code>PSMatch</code> package manual page
(<code><a href="../reference/PSMatch.html">?PSMatch</a></code>) and references therein.</p>
<p>To illustrate this vignette, we will import and merge raw and
identification data from the <em><a href="https://bioconductor.org/packages/3.22/msdata" class="external-link">msdata</a></em>.
For details about this section, please visit the <a href="https://rformassspectrometry.github.io/Spectra/articles/Spectra.html" class="external-link">Spectra</a>
package webpage.</p>
<p>Load the raw MS data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">spf</span> <span class="op">&lt;-</span> <span class="fu">msdata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msdata/man/proteomics.html" class="external-link">proteomics</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"2014"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/__w/_temp/Library/msdata/proteomics/TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML.gz"</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RforMassSpectrometry/Spectra" class="external-link">Spectra</a></span><span class="op">)</span></span>
<span><span class="va">sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Spectra/man/Spectra.html" class="external-link">Spectra</a></span><span class="op">(</span><span class="va">spf</span><span class="op">)</span></span></code></pre></div>
<p>Load the identification data:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">idf</span> <span class="op">&lt;-</span> <span class="fu">msdata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msdata/man/proteomics.html" class="external-link">ident</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"2014"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/__w/_temp/Library/msdata/ident/TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzid"</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PSM.html">PSM</a></span><span class="op">(</span><span class="va">idf</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/filterPSMs.html">filterPSMs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Starting with 5802 PSMs:</span></span></code></pre>
<pre><code><span><span class="co">## Removed 2896 decoy hits.</span></span></code></pre>
<pre><code><span><span class="co">## Removed 155 PSMs with rank &gt; 1.</span></span></code></pre>
<pre><code><span><span class="co">## Removed 85 shared peptides.</span></span></code></pre>
<pre><code><span><span class="co">## 2666 PSMs left.</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span></span></code></pre></div>
<pre><code><span><span class="co">## PSM with 2666 rows and 35 columns.</span></span>
<span><span class="co">## names(35): sequence spectrumID ... subReplacementResidue subLocation</span></span></code></pre>
<p>Merge both:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Spectra/man/combineSpectra.html" class="external-link">joinSpectraData</a></span><span class="op">(</span><span class="va">sp</span>, <span class="va">id</span>, by.x <span class="op">=</span> <span class="st">"spectrumId"</span>, by.y <span class="op">=</span> <span class="st">"spectrumID"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in joinSpectraData(sp, id, by.x = "spectrumId", by.y = "spectrumID"):</span></span>
<span><span class="co">## Duplicates found in the 'y' key. Only last instance will be kept!</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sp</span></span></code></pre></div>
<pre><code><span><span class="co">## MSn data (Spectra) with 7534 spectra in a MsBackendMzR backend:</span></span>
<span><span class="co">##        msLevel     rtime scanIndex</span></span>
<span><span class="co">##      &lt;integer&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">## 1            1    0.4584         1</span></span>
<span><span class="co">## 2            1    0.9725         2</span></span>
<span><span class="co">## 3            1    1.8524         3</span></span>
<span><span class="co">## 4            1    2.7424         4</span></span>
<span><span class="co">## 5            1    3.6124         5</span></span>
<span><span class="co">## ...        ...       ...       ...</span></span>
<span><span class="co">## 7530         2   3600.47      7530</span></span>
<span><span class="co">## 7531         2   3600.83      7531</span></span>
<span><span class="co">## 7532         2   3601.18      7532</span></span>
<span><span class="co">## 7533         2   3601.57      7533</span></span>
<span><span class="co">## 7534         2   3601.98      7534</span></span>
<span><span class="co">##  ... 68 more variables/columns.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## file(s):</span></span>
<span><span class="co">## TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML.gz</span></span></code></pre>
<p>In this example, we are going to focus the MS2 scan with index 1158
and its parent MS1 scan (index 1148, selected automatically with the <a href="https://rformassspectrometry.github.io/Spectra/reference/MsBackend.html" class="external-link">filterPrecursorScan()</a>
function).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sp1158</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">filterPrecursorScan</a></span><span class="op">(</span><span class="va">sp</span>, <span class="fl">1158</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Spectra/man/spectra-plotting.html" class="external-link">plotSpectra</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">400</span>, <span class="fl">600</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">precursorMz</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"red"</span>, lty <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Fragments_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="calculating-fragment-ions">Calculating fragment ions<a class="anchor" aria-label="anchor" href="#calculating-fragment-ions"></a>
</h2>
<p>The MS2 scan was matched to the sequence <code>SCALITDGR</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sp1158</span><span class="op">$</span><span class="va">sequence</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] NA          "SCALITDGR"</span></span></code></pre>
<p>The <code><a href="../reference/calculateFragments.html">calculateFragments()</a></code> simply takes a peptide sequence
as input and returns a <code>data.frame</code> with the fragment
sequences, M/Z, ion type, charge, position and the peptide sequence of
the parent ion.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/calculateFragments.html">calculateFragments</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">$</span><span class="va">sequence</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fixed modifications used: C=57.02146</span></span>
<span><span class="co">## Variable modifications used: None</span></span></code></pre>
<pre><code><span><span class="co">##           mz ion type pos z      seq   peptide</span></span>
<span><span class="co">## 1   88.03931  b1    b   1 1        S SCALITDGR</span></span>
<span><span class="co">## 2  248.06996  b2    b   2 1       SC SCALITDGR</span></span>
<span><span class="co">## 3  319.10707  b3    b   3 1      SCA SCALITDGR</span></span>
<span><span class="co">## 4  432.19113  b4    b   4 1     SCAL SCALITDGR</span></span>
<span><span class="co">## 5  545.27519  b5    b   5 1    SCALI SCALITDGR</span></span>
<span><span class="co">## 6  646.32287  b6    b   6 1   SCALIT SCALITDGR</span></span>
<span><span class="co">## 7  761.34981  b7    b   7 1  SCALITD SCALITDGR</span></span>
<span><span class="co">## 8  818.37127  b8    b   8 1 SCALITDG SCALITDGR</span></span>
<span><span class="co">## 9  175.11895  y1    y   1 1        R SCALITDGR</span></span>
<span><span class="co">## 10 232.14041  y2    y   2 1       GR SCALITDGR</span></span>
<span><span class="co">## 11 347.16735  y3    y   3 1      DGR SCALITDGR</span></span>
<span><span class="co">## 12 448.21503  y4    y   4 1     TDGR SCALITDGR</span></span>
<span><span class="co">## 13 561.29909  y5    y   5 1    ITDGR SCALITDGR</span></span>
<span><span class="co">## 14 674.38315  y6    y   6 1   LITDGR SCALITDGR</span></span>
<span><span class="co">## 15 745.42026  y7    y   7 1  ALITDGR SCALITDGR</span></span>
<span><span class="co">## 16 905.45091  y8    y   8 1 CALITDGR SCALITDGR</span></span>
<span><span class="co">## 17 743.33924 b7_   b_   7 1  SCALITD SCALITDGR</span></span>
<span><span class="co">## 18 800.36070 b8_   b_   8 1 SCALITDG SCALITDGR</span></span>
<span><span class="co">## 19 329.15679 y3_   y_   3 1      DGR SCALITDGR</span></span>
<span><span class="co">## 20 543.28853 y5_   y_   5 1    ITDGR SCALITDGR</span></span>
<span><span class="co">## 21 656.37259 y6_   y_   6 1   LITDGR SCALITDGR</span></span>
<span><span class="co">## 22 727.40970 y7_   y_   7 1  ALITDGR SCALITDGR</span></span>
<span><span class="co">## 23 887.44035 y8_   y_   8 1 CALITDGR SCALITDGR</span></span>
<span><span class="co">## 24 157.10839 y1_   y_   1 1        R SCALITDGR</span></span>
<span><span class="co">## 25 214.12985 y2_   y_   2 1       GR SCALITDGR</span></span>
<span><span class="co">## 26 430.20447 y4_   y_   4 1     TDGR SCALITDGR</span></span></code></pre>
<p>The function also allows to generate fragment sequences with fixed
and/or variable modifications. By default,
<code>fixed_modifications = c(C = 57.02146)</code> for
carbamidomethylation of cysteine.</p>
<p>With variable modifications, multiple sets of fragments are
generated. The fragments can be traced to their parent ion by checking
the <code>peptide</code> column. A fragment can have multiple
modifications.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/calculateFragments.html">calculateFragments</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">$</span><span class="va">sequence</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>                   fixed_modifications <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                   variable_modifications <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>C <span class="op">=</span> <span class="fl">57.02146</span>, </span>
<span>                                              T <span class="op">=</span> <span class="fl">79.966</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fixed modifications used: None</span></span>
<span><span class="co">## Variable modifications used: C=57.02146, T=79.966</span></span></code></pre>
<pre><code><span><span class="co">##            mz ion type pos z      seq                     peptide</span></span>
<span><span class="co">## 1    88.03931  b1    b   1 1        S                   SCALITDGR</span></span>
<span><span class="co">## 2   191.04850  b2    b   2 1       SC                   SCALITDGR</span></span>
<span><span class="co">## 3   262.08561  b3    b   3 1      SCA                   SCALITDGR</span></span>
<span><span class="co">## 4   375.16967  b4    b   4 1     SCAL                   SCALITDGR</span></span>
<span><span class="co">## 5   488.25373  b5    b   5 1    SCALI                   SCALITDGR</span></span>
<span><span class="co">## 6   589.30141  b6    b   6 1   SCALIT                   SCALITDGR</span></span>
<span><span class="co">## 7   704.32835  b7    b   7 1  SCALITD                   SCALITDGR</span></span>
<span><span class="co">## 8   761.34981  b8    b   8 1 SCALITDG                   SCALITDGR</span></span>
<span><span class="co">## 9   175.11895  y1    y   1 1        R                   SCALITDGR</span></span>
<span><span class="co">## 10  232.14041  y2    y   2 1       GR                   SCALITDGR</span></span>
<span><span class="co">## 11  347.16735  y3    y   3 1      DGR                   SCALITDGR</span></span>
<span><span class="co">## 12  448.21503  y4    y   4 1     TDGR                   SCALITDGR</span></span>
<span><span class="co">## 13  561.29909  y5    y   5 1    ITDGR                   SCALITDGR</span></span>
<span><span class="co">## 14  674.38315  y6    y   6 1   LITDGR                   SCALITDGR</span></span>
<span><span class="co">## 15  745.42026  y7    y   7 1  ALITDGR                   SCALITDGR</span></span>
<span><span class="co">## 16  848.42945  y8    y   8 1 CALITDGR                   SCALITDGR</span></span>
<span><span class="co">## 17  686.31778 b7_   b_   7 1  SCALITD                   SCALITDGR</span></span>
<span><span class="co">## 18  743.33924 b8_   b_   8 1 SCALITDG                   SCALITDGR</span></span>
<span><span class="co">## 19  329.15679 y3_   y_   3 1      DGR                   SCALITDGR</span></span>
<span><span class="co">## 20  543.28853 y5_   y_   5 1    ITDGR                   SCALITDGR</span></span>
<span><span class="co">## 21  656.37259 y6_   y_   6 1   LITDGR                   SCALITDGR</span></span>
<span><span class="co">## 22  727.40970 y7_   y_   7 1  ALITDGR                   SCALITDGR</span></span>
<span><span class="co">## 23  830.41889 y8_   y_   8 1 CALITDGR                   SCALITDGR</span></span>
<span><span class="co">## 24  157.10839 y1_   y_   1 1        R                   SCALITDGR</span></span>
<span><span class="co">## 25  214.12985 y2_   y_   2 1       GR                   SCALITDGR</span></span>
<span><span class="co">## 26  430.20447 y4_   y_   4 1     TDGR                   SCALITDGR</span></span>
<span><span class="co">## 27   88.03931  b1    b   1 1        S         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 28  248.06996  b2    b   2 1       SC         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 29  319.10707  b3    b   3 1      SCA         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 30  432.19113  b4    b   4 1     SCAL         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 31  545.27519  b5    b   5 1    SCALI         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 32  646.32287  b6    b   6 1   SCALIT         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 33  761.34981  b7    b   7 1  SCALITD         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 34  818.37127  b8    b   8 1 SCALITDG         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 35  175.11895  y1    y   1 1        R         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 36  232.14041  y2    y   2 1       GR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 37  347.16735  y3    y   3 1      DGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 38  448.21503  y4    y   4 1     TDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 39  561.29909  y5    y   5 1    ITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 40  674.38315  y6    y   6 1   LITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 41  745.42026  y7    y   7 1  ALITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 42  905.45091  y8    y   8 1 CALITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 43  743.33924 b7_   b_   7 1  SCALITD         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 44  800.36070 b8_   b_   8 1 SCALITDG         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 45  329.15679 y3_   y_   3 1      DGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 46  543.28853 y5_   y_   5 1    ITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 47  656.37259 y6_   y_   6 1   LITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 48  727.40970 y7_   y_   7 1  ALITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 49  887.44035 y8_   y_   8 1 CALITDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 50  157.10839 y1_   y_   1 1        R         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 51  214.12985 y2_   y_   2 1       GR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 52  430.20447 y4_   y_   4 1     TDGR         SC[57.02146]ALITDGR</span></span>
<span><span class="co">## 53   88.03931  b1    b   1 1        S           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 54  191.04850  b2    b   2 1       SC           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 55  262.08561  b3    b   3 1      SCA           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 56  375.16967  b4    b   4 1     SCAL           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 57  488.25373  b5    b   5 1    SCALI           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 58  669.26741  b6    b   6 1   SCALIT           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 59  784.29435  b7    b   7 1  SCALITD           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 60  841.31581  b8    b   8 1 SCALITDG           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 61  175.11895  y1    y   1 1        R           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 62  232.14041  y2    y   2 1       GR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 63  347.16735  y3    y   3 1      DGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 64  528.18103  y4    y   4 1     TDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 65  641.26509  y5    y   5 1    ITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 66  754.34915  y6    y   6 1   LITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 67  825.38626  y7    y   7 1  ALITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 68  928.39545  y8    y   8 1 CALITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 69  766.28378 b7_   b_   7 1  SCALITD           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 70  823.30524 b8_   b_   8 1 SCALITDG           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 71  329.15679 y3_   y_   3 1      DGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 72  623.25453 y5_   y_   5 1    ITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 73  736.33859 y6_   y_   6 1   LITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 74  807.37570 y7_   y_   7 1  ALITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 75  910.38489 y8_   y_   8 1 CALITDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 76  157.10839 y1_   y_   1 1        R           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 77  214.12985 y2_   y_   2 1       GR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 78  510.17047 y4_   y_   4 1     TDGR           SCALIT[79.966]DGR</span></span>
<span><span class="co">## 79   88.03931  b1    b   1 1        S SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 80  248.06996  b2    b   2 1       SC SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 81  319.10707  b3    b   3 1      SCA SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 82  432.19113  b4    b   4 1     SCAL SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 83  545.27519  b5    b   5 1    SCALI SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 84  726.28887  b6    b   6 1   SCALIT SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 85  841.31581  b7    b   7 1  SCALITD SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 86  898.33727  b8    b   8 1 SCALITDG SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 87  175.11895  y1    y   1 1        R SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 88  232.14041  y2    y   2 1       GR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 89  347.16735  y3    y   3 1      DGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 90  528.18103  y4    y   4 1     TDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 91  641.26509  y5    y   5 1    ITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 92  754.34915  y6    y   6 1   LITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 93  825.38626  y7    y   7 1  ALITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 94  985.41691  y8    y   8 1 CALITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 95  823.30524 b7_   b_   7 1  SCALITD SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 96  880.32670 b8_   b_   8 1 SCALITDG SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 97  329.15679 y3_   y_   3 1      DGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 98  623.25453 y5_   y_   5 1    ITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 99  736.33859 y6_   y_   6 1   LITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 100 807.37570 y7_   y_   7 1  ALITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 101 967.40635 y8_   y_   8 1 CALITDGR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 102 157.10839 y1_   y_   1 1        R SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 103 214.12985 y2_   y_   2 1       GR SC[57.02146]ALIT[79.966]DGR</span></span>
<span><span class="co">## 104 510.17047 y4_   y_   4 1     TDGR SC[57.02146]ALIT[79.966]DGR</span></span></code></pre>
<p>Additional parameters can limit the maximum number of allowed
modifications, the type of ions produced or the charge applied. See
<code><a href="../reference/calculateFragments.html">?calculateFragments</a></code> for more details on those.</p>
</div>
<div class="section level2">
<h2 id="visualising-fragment-ions">Visualising fragment ions<a class="anchor" aria-label="anchor" href="#visualising-fragment-ions"></a>
</h2>
<p>We can now visualise these fragments directly on the MS spectrum.
Let’s first visualise the spectrum as is:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Spectra/man/spectra-plotting.html" class="external-link">plotSpectra</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="Fragments_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p><code><a href="../reference/plotSpectraPTM.html">plotSpectraPTM()</a></code> allows a more in depth visualisation of
a PSM by providing a delta mass plot of matched fragments and a direct
visualisation of matched b- and y-ion fragment sequences.</p>
<p>Labels are automatically applied based on the <code>sequence</code>
defined in the <code>spectraVariables</code> with carbamidomethylation
of cysteines set as default fixed modification.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">dataOrigin</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"TMT_Erwinia"</span> <span class="co">## Reduces the mzspec text</span></span>
<span><span class="fu"><a href="../reference/plotSpectraPTM.html">plotSpectraPTM</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>               main <span class="op">=</span> <span class="st">"Scan 1158 with carbamidomethylation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Fragments_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>More importantly, <code><a href="../reference/plotSpectraPTM.html">plotSpectraPTM()</a></code> allows to visualise
and compare the matches with different modifications. For instance,
there is a better match when carbamidomethylation of cysteines is
applied (as above) compared to no modifications at all.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpectraPTM.html">plotSpectraPTM</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>               fixed_modifications <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>               variable_modifications <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>               main <span class="op">=</span> <span class="st">"Scan 1158 without modifications"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Fragments_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>As glycine has the same mass as carbamidomethylation, the b7 and b8
ions are overlapping in both spectra).</p>
<p>Both plots can be created at once using the
<code>variable_modifications</code> parameter.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpectraPTM.html">plotSpectraPTM</a></span><span class="op">(</span><span class="va">sp1158</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>               fixed_modifications <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>               variable_modifications <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>C <span class="op">=</span> <span class="fl">57.02146</span><span class="op">)</span>,</span>
<span>               deltaMz <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Fragments_files/figure-html/unnamed-chunk-12-1.png" width="1000px"></p>
<p>For more details on what <code><a href="../reference/plotSpectraPTM.html">plotSpectraPTM()</a></code> can do, run
<code><a href="../reference/plotSpectraPTM.html">?plotSpectraPTM</a></code>.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] Spectra_1.19.4      BiocParallel_1.43.4 PSMatch_1.13.3     </span></span>
<span><span class="co">## [4] S4Vectors_0.47.0    BiocGenerics_0.55.1 generics_0.1.4     </span></span>
<span><span class="co">## [7] BiocStyle_2.37.0   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.39.1 xfun_0.52                  </span></span>
<span><span class="co">##  [3] bslib_0.9.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [5] QFeatures_1.19.2            Biobase_2.69.0             </span></span>
<span><span class="co">##  [7] lattice_0.22-7              vctrs_0.6.5                </span></span>
<span><span class="co">##  [9] tools_4.5.1                 parallel_4.5.1             </span></span>
<span><span class="co">## [11] tibble_3.3.0                cluster_2.1.8.1            </span></span>
<span><span class="co">## [13] pkgconfig_2.0.3             BiocBaseUtils_1.11.2       </span></span>
<span><span class="co">## [15] Matrix_1.7-3                desc_1.4.3                 </span></span>
<span><span class="co">## [17] lifecycle_1.0.4             compiler_4.5.1             </span></span>
<span><span class="co">## [19] stringr_1.5.1               textshaping_1.0.1          </span></span>
<span><span class="co">## [21] Seqinfo_0.99.2              codetools_0.2-20           </span></span>
<span><span class="co">## [23] ncdf4_1.24                  clue_0.3-66                </span></span>
<span><span class="co">## [25] htmltools_0.5.8.1           sass_0.4.10                </span></span>
<span><span class="co">## [27] yaml_2.3.10                 lazyeval_0.2.2             </span></span>
<span><span class="co">## [29] pillar_1.11.0               pkgdown_2.1.3.9000         </span></span>
<span><span class="co">## [31] crayon_1.5.3                jquerylib_0.1.4            </span></span>
<span><span class="co">## [33] tidyr_1.3.1                 MASS_7.3-65                </span></span>
<span><span class="co">## [35] cachem_1.1.0                DelayedArray_0.35.2        </span></span>
<span><span class="co">## [37] MetaboCoreUtils_1.17.1      abind_1.4-8                </span></span>
<span><span class="co">## [39] tidyselect_1.2.1            digest_0.6.37              </span></span>
<span><span class="co">## [41] stringi_1.8.7               purrr_1.1.0                </span></span>
<span><span class="co">## [43] dplyr_1.1.4                 reshape2_1.4.4             </span></span>
<span><span class="co">## [45] bookdown_0.43               fastmap_1.2.0              </span></span>
<span><span class="co">## [47] grid_4.5.1                  cli_3.6.5                  </span></span>
<span><span class="co">## [49] SparseArray_1.9.1           magrittr_2.0.3             </span></span>
<span><span class="co">## [51] S4Arrays_1.9.1              rmarkdown_2.29             </span></span>
<span><span class="co">## [53] XVector_0.49.0              matrixStats_1.5.0          </span></span>
<span><span class="co">## [55] igraph_2.1.4                ragg_1.4.0                 </span></span>
<span><span class="co">## [57] evaluate_1.0.4              knitr_1.50                 </span></span>
<span><span class="co">## [59] GenomicRanges_1.61.1        IRanges_2.43.0             </span></span>
<span><span class="co">## [61] MultiAssayExperiment_1.35.6 rlang_1.1.6                </span></span>
<span><span class="co">## [63] Rcpp_1.1.0                  glue_1.8.0                 </span></span>
<span><span class="co">## [65] mzR_2.43.0                  BiocManager_1.30.26        </span></span>
<span><span class="co">## [67] msdata_0.49.0               jsonlite_2.0.0             </span></span>
<span><span class="co">## [69] AnnotationFilter_1.33.0     R6_2.6.1                   </span></span>
<span><span class="co">## [71] plyr_1.8.9                  MatrixGenerics_1.21.0      </span></span>
<span><span class="co">## [73] systemfonts_1.2.3           fs_1.6.6                   </span></span>
<span><span class="co">## [75] ProtGenerics_1.41.0         MsCoreUtils_1.21.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto, Johannes Rainer, Sebastian Gibb.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.9000.</p>
</div>

      </footer>
</div>






  </body>
</html>
